RevenueCatUI/CustomerCenter/Views/RestorePurchasesAlert.swift

```diff
@@ -32,23 +32,50 @@ struct RestorePurchasesAlert: ViewModifier {
     var openURL
 
     @EnvironmentObject private var customerCenterViewModel: CustomerCenterViewModel
+    @StateObject private var viewModel: RestorePurchasesAlertViewModel
 
     @State
-    private var alertType: AlertType
+    private var alertType: RestorePurchasesAlertViewModel.AlertType
```

Comment:
These `@State` and `@StateObject` don't make sense here. The variables are ok, but the property wrapper does not.

`alertType` should be passed a as `@Binding` like `isPresented`.  The view using the modifier is the one driving the State, not the ViewModifier which must be stateless.